### PR TITLE
Fixing MultiDenseVectorScriptDocValuesTests tests

### LINKED ISSUE #116862: [CI] MultiDenseVectorScriptDocValuesTests testFloatGetVectorValueAndGetMagnitude failing
**Build Scans:**
- [elasticsearch-periodic-platform-support #4849 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/nxljtssl2tcwy)
- [elasticsearch-periodic-platform-support #4849 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/7uzbo3twwtqw4)
- [elasticsearch-periodic-platform-support #4849 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/q7qab2cr5owua)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.vectors.MultiDenseVectorScriptDocValuesTests.testFloatGetVectorValueAndGetMagnitude" -Dtests.seed=C448BDA2B40AB443 -Dtests.locale=mas -Dtests.timezone=Pacific/Chuuk -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.vectors.MultiDenseVectorScriptDocValuesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testFloatGetVectorValueAndGetMagnitude),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.junit.internal.ArrayComparisonFailure: arrays first differed at element [0]; expected:<1.0> but was:<4.6006E-41>
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testFloatGetVectorValueAndGetMagnitude
- [main] 3 failures in test testFloatGetVectorValueAndGetMagnitude (20.0% fail rate in 15 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### LINKED ISSUE #116863: [CI] MultiDenseVectorScriptDocValuesTests testFloatGetVectorValueAndGetMagnitude failing
**Build Scans:**
- [elasticsearch-periodic-platform-support #4854 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/l3bciummidpt2)
- [elasticsearch-periodic-platform-support #4854 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/x26kgde2f5rp2)
- [elasticsearch-periodic-platform-support #4854 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/q3fghwjd6g6y4)
- [elasticsearch-intake #12947 / part1](https://gradle-enterprise.elastic.co/s/w5juylfmcch5u)
- [elasticsearch-periodic-platform-support #4849 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/s6gxguejrhnas)
- [elasticsearch-periodic-platform-support #4849 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/cufxe22ivrgwi)
- [elasticsearch-periodic #4874 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/oqgp4qhzqrzp6)
- [elasticsearch-periodic-platform-support #4849 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/azefnhq4dqnle)
- [elasticsearch-periodic-platform-support #4849 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/xaozxuadebdnq)
- [elasticsearch-periodic-platform-support #4849 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/gzw7qzcpsigys)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.vectors.MultiDenseVectorScriptDocValuesTests.testFloatGetVectorValueAndGetMagnitude" -Dtests.seed=F41D1F3684F2824C -Dtests.locale=seh -Dtests.timezone=AST -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.vectors.MultiDenseVectorScriptDocValuesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testFloatGetVectorValueAndGetMagnitude),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.junit.internal.ArrayComparisonFailure: arrays first differed at element [0]; expected:<1.0> but was:<4.6006E-41>
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testFloatGetVectorValueAndGetMagnitude
- [main] 2 consecutive failures in step rhel-7_platform-support-unix
- [main] 2 consecutive failures in step ubuntu-2204_platform-support-unix
- [main] 2 consecutive failures in step rhel-8_platform-support-unix
- [main] 14 failures in test testFloatGetVectorValueAndGetMagnitude (30.4% fail rate in 46 executions)
- [main] 2 failures in step rhel-7_platform-support-unix (100.0% fail rate in 2 executions)
- [main] 2 failures in step ubuntu-2204_platform-support-unix (100.0% fail rate in 2 executions)
- [main] 2 failures in step rhel-8_platform-support-unix (100.0% fail rate in 2 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by benwtrent**
This fixes two test issues:

 - 1. Now the tests skip if the multi_dense_vector feature isn't enabled
 - 2. fixes silly bwc testing where we were testing for big-endian floats, which aren't possible.

closes: https://github.com/elastic/elasticsearch/issues/116862
closes: https://github.com/elastic/elasticsearch/issues/116863

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValuesTests.java [hunk 1]
```diff
@@ -18,46 +18,48 @@
...
 import java.util.Iterator;
-import java.util.List;
 
 import static org.hamcrest.Matchers.containsString;
 
 public class MultiDenseVectorScriptDocValuesTests extends ESTestCase {
 
+    @BeforeClass
+    public static void setup() {
+        assumeTrue("Requires multi-dense vector support", MultiDenseVectorFieldMapper.FEATURE_FLAG.isEnabled());
```
Comment (mayya-sharipova) [inline server/src/test/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValuesTests.java:34]: What happens when we remove this feature flag? Should we rather reply on more stable `SearchCapabilities`?

**Review by mayya-sharipova — APPROVED** [review]
Thanks Ben

File: server/src/test/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValuesTests.java (hunk 1 above)
Comment (benwtrent) [reply server/src/test/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValuesTests.java:34]: @mayya-sharipova feature flag is there as the initial work is all done over snapshot builds or specifically set behind a feature flag. This is OK here and if we ever remove the flag, that means its on and these setups can be removed. 

The capability checks are only for yaml test skipping to my knowledge.

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/116976)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**